### PR TITLE
cookies not compressed

### LINKED ISSUE #21455: Large Cookie Size Issue Due to IDataSerializer not using gzip compression : Documentation needed
I have a .NET 4.5.2 web app that uses ASP.NET Identity for authentication. There is large data in one of the claims (Serialized json) which went un-noticed until we tried to share the cookie with asp.net core application.

Basically, the IDataSerializer<AuthenticationTicket> implementation used by Katana has gzip compression while writing cookie. Same is absent in Interop serializer and asp.net core serializer. This results in large cookie which mandates chunking.

I would like to know reason behind this change(no compression in asp.net core vs gzip in katana/owin). Also if we could highlight this change in the documentation, it would help developers to comprehend drastic change in cookie size after enabling AspNetTicketDataFormat

# Software versions

Check the .NET target framework(s) being used, and include the version number(s).

* [ ] .NET Core 2.2
* [ ] .NET Framework 4.5.2

If using the .NET Core SDK, include `dotnet --info` output. If using .NET Framework without the .NET Core SDK, include info from Visual Studio's **Help** > **About Microsoft Visual Studio** dialog.

<details>
<summary><strong>dotnet --info output</strong> or <strong>About VS info</strong></summary>

```console
<replace>
```
</details>

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #21455

[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/migration/proper-to-2x/?view=aspnetcore-1.0&branch=pr-en-us-21571#cookies-are-not-compressed-in-aspnet-core)
[2nd review](https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/app-state?view=aspnetcore-6.0&branch=pr-en-us-21571#cookies)

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@vaibhavramdasi added your suggestion

File: aspnetcore/fundamentals/app-state.md [hunk 1]
```diff
@@ -42,6 +42,8 @@ Cookies are often used for personalization, where content is customized for a kn
 
 See the [European Union General Data Protection Regulations (GDPR)](https://ec.europa.eu/info/law/law-topic/data-protection) when issuing cookies and dealing with privacy concerns. For more information, see [General Data Protection Regulation (GDPR) support in ASP.NET Core](xref:security/gdpr).
 
+[!INCLUDE[](~/includes/cookies-not-compressed.md)]
+
```
Comment (Tratcher) [inline aspnetcore/fundamentals/app-state.md:46]: This differentiation only applies to auth cookies, not app state cookies.
```suggestion
```

File: aspnetcore/includes/cookies-not-compressed.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+no-loc: [appsettings.json, "ASP.NET Core Identity", cookie, Cookie, Blazor, "Blazor Server", "Blazor WebAssembly", "Identity", "Let's Encrypt", Razor, SignalR]
+---
+For security reasons, cookies are not compressed in ASP.NET Core. When using authorization cookies, developers should minimize the number of claim information included to just that necessary for their needs.
```
Comment (Tratcher) [inline aspnetcore/includes/cookies-not-compressed.md:4]:
```suggestion
For security reasons, authentication cookies are not compressed in ASP.NET Core. When using authentication cookies, developers should minimize the number of claim information included to just that necessary for their needs.
```

File: aspnetcore/migration/proper-to-2x/index.md [hunk 3]
```diff
@@ -154,6 +154,10 @@ For example, an image asset in the *wwwroot/images* folder is accessible to the
 
 [Multi-value cookies](xref:System.Web.HttpCookie.Values) aren't supported in ASP.NET Core. Create one cookie per value.
 
+## Cookies are not compressed in ASP.NET Core
```
Comment (Tratcher) [inline aspnetcore/migration/proper-to-2x/index.md:157]:
```suggestion
## Authentication cookies are not compressed in ASP.NET Core
```

**Review by Tratcher — APPROVED** [review]

**Pull request merged**